pyacq/core/rpc/tests/rpc_benchmark.py

```diff
@@ -0,0 +1,68 @@
+import time, threading
+from pyacq.core import rpc
+
+def test_poingrate(cli, dur=10.0):
+    start = time.time()
+    count = 0
+    while time.time() < start + dur:
+        assert cli.send('ping', sync='sync') == 'pong'
+        count += 1
+        
+    print("Ping rate: %0.0f/sec" % (count/dur))
+
+
+def test_async_poingrate(cli, dur=10.0, buffer=500):
+    start = time.time()
+    count = 0
+    futs = []
+    for i in range(buffer):
+        futs.append(cli.send('ping', sync='async'))
+    while time.time() < start + dur:
+        futs.append(cli.send('ping', sync='async'))
+        assert futs.pop(0).result() == 'pong'
+        count += 1
+        
+    print("Async ping rate: %0.0f/sec" % (count/dur))
+
+    
+    
+
+# thread, inproc
+print("=========== inproc to thread ============")
+server = rpc.RPCServer('inproc://testserver')
+thread = threading.Thread(target=server.run_forever, daemon=True)
+thread.start()
+
+cli = rpc.RPCClient(server.address)
+test_poingrate(cli)
+test_async_poingrate(cli)
+
+cli.close()
+
+
+# thread, tcp
+print("=========== tcp to thread ============")
+server = rpc.RPCServer('tcp://*:*')
+thread = threading.Thread(target=server.run_forever, daemon=True)
+thread.start()
+
+cli = rpc.RPCClient(server.address)
+test_poingrate(cli)
+test_async_poingrate(cli)
+
+cli.close()
+
+
+# process
+print("=========== TCP to spawned process ============")
+proc = rpc.ProcessSpawner()
+test_poingrate(proc.client)
+test_async_poingrate(proc.client)
+
+
+# process
+print("=========== TCP to spawned Qt process ============")
+proc = rpc.ProcessSpawner(qt=True)
+test_poingrate(proc.client)
+test_async_poingrate(proc.client)
+
```

Comment:
On my machine I get this:

=========== inproc to thread ============
Ping rate: 2644/sec
Async ping rate: 2786/sec
=========== tcp to thread ============
Ping rate: 1672/sec
Async ping rate: 2697/sec
=========== TCP to spawned process ============
Ping rate: 1652/sec
Async ping rate: 4687/sec
=========== TCP to spawned Qt process ============
Ping rate: 1268/sec
Async ping rate: 4396/sec
WARNING:pyacq.core.rpc.server:RPCServer exiting without close()!
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/samuel/Documents/projet/pyacq-0.2/pyacq/core/rpc/client.py", line 306, in _read_and_process_all
    self._read_and_process_one(timeout=0)
  File "/home/samuel/Documents/projet/pyacq-0.2/pyacq/core/rpc/client.py", line 294, in _read_and_process_one
    self._socket.setsockopt(zmq.RCVTIMEO, timeout)
  File "zmq/backend/cython/socket.pyx", line 308, in zmq.backend.cython.socket.Socket.set (zmq/backend/cython/socket.c:3163)
  File "zmq/backend/cython/socket.pyx", line 96, in zmq.backend.cython.socket._check_closed (zmq/backend/cython/socket.c:1324)
zmq.error.ZMQError: Socket operation on non-socket
WARNING:pyacq.core.rpc.server:RPCServer exiting without close()!
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/samuel/Documents/projet/pyacq-0.2/pyacq/core/rpc/client.py", line 306, in _read_and_process_all
    self._read_and_process_one(timeout=0)
  File "/home/samuel/Documents/projet/pyacq-0.2/pyacq/core/rpc/client.py", line 294, in _read_and_process_one
    self._socket.setsockopt(zmq.RCVTIMEO, timeout)
  File "zmq/backend/cython/socket.pyx", line 308, in zmq.backend.cython.socket.Socket.set (zmq/backend/cython/socket.c:3163)
  File "zmq/backend/cython/socket.pyx", line 96, in zmq.backend.cython.socket._check_closed (zmq/backend/cython/socket.c:1324)
zmq.error.ZMQError: Socket operation on non-socket